framework/src/userobjects/TimedSubdomainModifier.C

```diff
@@ -0,0 +1,298 @@
+
+  return params;
+}
+
+TimedSubdomainModifier::TimedSubdomainModifier(const InputParameters & parameters)
+  : TimedElementSubdomainModifier(parameters)
+{
+  // determine function arguments
+
+  int bFromData_File =
+      isParamValid("data_file") + isParamValid("header") + isParamValid("delimiter") +
+      isParamValid("comment") + isParamValid("time_column_index") +
+      isParamValid("blocks_from_column_index") + isParamValid("blocks_to_column_index") +
+      isParamValid("time_column_text") + isParamValid("blocks_from_column_text") +
+      isParamValid("blocks_to_column_text");
+
+  int bFromData_File_NeedsHeader = isParamValid("time_column_text") +
+                                   isParamValid("blocks_from_column_text") +
+                                   isParamValid("blocks_to_column_text");
+
+  int bFromParameters =
+      isParamValid("times") + isParamValid("blocks_from") + isParamValid("blocks_to");
+
+  // Check parameter set for inconsistencies
+  int nFrom = ((bFromData_File > 0) + (bFromParameters > 0));
+  if (nFrom != 1)
+  {
+    mooseError("Data on times and blocks must be provided either via a CSV file ('data_file' and "
+               "corresponding blocks), or via direct parameter input ('times', 'blocks_from', and "
+               "'blocks_to').");
+  };
+
+  if (bFromParameters > 0)
+  {
+    if (bFromParameters != 3)
+    {
+      mooseError("All parameters 'times', and 'blocks_from', and 'blocks_to' must be specified.");
+    };
+    buildFromParameters();
+  }
+  else if (bFromData_File > 0)
+  {
+    if ((isParamSetByUser("time_column_index") + isParamSetByUser("time_column_text")) > 1)
+      mooseError(
+          "The parameters 'time_column_index', and 'time_column_text' are mutual exclusive.");
+    if ((isParamSetByUser("blocks_from_column_index") +
+         isParamSetByUser("blocks_from_column_text")) > 1)
+      mooseError("The parameters 'blocks_from_column_index', and 'blocks_from_column_text' are "
+                 "mutual exclusive.");
+    if ((isParamSetByUser("blocks_to_column_index") + isParamSetByUser("blocks_to_column_text")) >
+        1)
+      mooseError("The parameters 'blocks_to_column_index', and 'blocks_to_column_text' are mutual "
+                 "exclusive.");
+    if ((bFromData_File_NeedsHeader > 0) && (isParamSetByUser("header") == 0))
+      mooseError("Header flag must be active if columns are to be found via header.");
+
+    buildFromFile();
+  }
+  else
+  {
+    mooseError("Unknown data source. Are you missing a parameter? Did you misspell one?");
+  };
+}
+
+void
+TimedSubdomainModifier::buildFromParameters()
+{
+
+  _times = this->template getParam<std::vector<Real>>("times");
+  size_t n = _times.size();
+
+  std::vector<std::string> _raw_from = getParam<std::vector<std::string>>("blocks_from");
+  std::vector<std::string> _raw_to = getParam<std::vector<std::string>>("blocks_to");
```

Comment:
```suggestion
  const auto raw_from = getParam<std::vector<SubdomainName>>("blocks_from");
  const auto raw_to = getParam<std::vector<SubdomainName>>("blocks_to");
```
leading _ is only for member variables